open_api_framework/conf/base.py

```diff
@@ -0,0 +1,662 @@
+# hatch to override it.
+CSRF_TRUSTED_ORIGINS = config(
+    "CSRF_TRUSTED_ORIGINS",
+    split=True,
+    default=[strip_protocol_from_origin(origin) for origin in CORS_ALLOWED_ORIGINS],
+)
+#
+# DJANGO-PRIVATES -- safely serve files after authorization
+#
+PRIVATE_MEDIA_ROOT = os.path.join(BASE_DIR, "private-media")
+PRIVATE_MEDIA_URL = "/private-media/"
+
+
+#
+# NOTIFICATIONS-API-COMMON
+#
+NOTIFICATIONS_DISABLED = config("NOTIFICATIONS_DISABLED", default=False)
+
+#
+# SENTRY - error monitoring
+#
+
+
+def init_sentry(before_send: Callable | None = None):
+    SENTRY_DSN = config("SENTRY_DSN", None)
+
+    if SENTRY_DSN:
+        SENTRY_CONFIG = {
+            "dsn": SENTRY_DSN,
+            "release": RELEASE or "RELEASE not set",
+            "environment": ENVIRONMENT,
+        }
+
+        # Allow projects to define their own before_send filters
+        if before_send:
+            SENTRY_CONFIG["before_send"] = before_send
+
+        sentry_sdk.init(
+            **SENTRY_CONFIG,
+            integrations=get_sentry_integrations(),
+            send_default_pii=True,
+        )
+
+
+#
+# CELERY
+#
+CELERY_BROKER_URL = config("CELERY_RESULT_BACKEND", "redis://localhost:6379/1")
+CELERY_RESULT_BACKEND = config("CELERY_RESULT_BACKEND", "redis://localhost:6379/1")
+
+
+#
+# DJANGO-ADMIN-INDEX
+#
+ADMIN_INDEX_SHOW_REMAINING_APPS_TO_SUPERUSERS = False
+ADMIN_INDEX_AUTO_CREATE_APP_GROUP = False
+
+#
+# Mozilla Django OIDC DB settings
+#
+OIDC_AUTHENTICATE_CLASS = "mozilla_django_oidc_db.views.OIDCAuthenticationRequestView"
+MOZILLA_DJANGO_OIDC_DB_CACHE = "oidc"
+MOZILLA_DJANGO_OIDC_DB_CACHE_TIMEOUT = 5 * 60
+
+#
+# Elastic APM
+#
+ELASTIC_APM_SERVER_URL = config("ELASTIC_APM_SERVER_URL", None)
+ELASTIC_APM = {
+    # FIXME this does change the default service name, because PROJECT_DIRNAME != PROJECT_NAME
+    "SERVICE_NAME": config(
+        "ELASTIC_APM_SERVICE_NAME", f"{PROJECT_DIRNAME} - {ENVIRONMENT}"
+    ),
+    "SECRET_TOKEN": config("ELASTIC_APM_SECRET_TOKEN", "default"),
+    "SERVER_URL": ELASTIC_APM_SERVER_URL,
+}
+if not ELASTIC_APM_SERVER_URL:
+    ELASTIC_APM["ENABLED"] = False
+    ELASTIC_APM["SERVER_URL"] = "http://localhost:8200"
+else:
+    MIDDLEWARE = ["elasticapm.contrib.django.middleware.TracingMiddleware"] + MIDDLEWARE
+    INSTALLED_APPS = INSTALLED_APPS + [
+        "elasticapm.contrib.django",
+    ]
+
+
+#
+# MAYKIN-2FA
+# Uses django-two-factor-auth under the hood, so relevant upstream package settings
+# apply too.
+#
+
+# we run the admin site monkeypatch instead.
+TWO_FACTOR_PATCH_ADMIN = False
+# add entries from AUTHENTICATION_BACKENDS that already enforce their own two-factor
+# auth, avoiding having some set up MFA again in the project.
+MAYKIN_2FA_ALLOW_MFA_BYPASS_BACKENDS = [
+    "mozilla_django_oidc_db.backends.OIDCAuthenticationBackend",
+]
+
+if config("DISABLE_2FA", default=False):  # pragma: no cover
+    MAYKIN_2FA_ALLOW_MFA_BYPASS_BACKENDS = AUTHENTICATION_BACKENDS
```

Comment:
Not 100% sure if we want this here, OF only has this in `dev.py`, but that way you wouldn't be able to disable 2FA for docker containers locally for example.

@annashamray thoughts?

Reply:
Alright, then I'll leave it in there :+1: